Replace boost_python parameterset reader with a python3 version

davidlange6 opened the pull request:
#### PR description:

- Remove boost_python version of cmsRun
- rename pybind11+python2 parameterset interface to FWCore/PythonParameterSet
- copy  FWCore/PythonParameterSet to  FWCore/PyDevParameterSet but with classes in different namespace for a pybind11+python3 parameterset interface

Despite the big changes, there is NO change to cmsRun itself aside from some package renaming. And, the big chunk of new code is a direct copy of old code - this duplication (which I don't see how to avoid, but interested in any ideas) will go away when we drop support for python2 in cmsRun

#### PR validation:

unit tests and basic runTheMatrix ok.
#### if this PR is a backport please specify the original PR:

davidlange6 committed:
remove boost python pset

davidlange6 committed:
add python3 cmsRun as PyDevParameterSet. Move everything to use PythonParameterSet

davidlange6 committed:
move unit test to ParameterSetReader interface

davidlange6 commented:
please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26377/9121

- This PR adds an extra 56KB to repository

- Found files with invalid states:
  - FWCore/PythonParameterSet/BuildFile.xml:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/makeprocess_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/makepset_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/BuildFile.xml:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/python/test/testTask_cff.py:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/ufip.txt:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/fip.txt:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/processbuilder_t.cppunit.cpp:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/readpsetsfrom_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f


Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26377/9121/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26377/9121/git-diff.patch | patch -p1`

You can run `scram build code-checks` to apply code checks directly

davidlange6 committed:
clang tidy fix

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26377/9122

- This PR adds an extra 56KB to repository

- Found files with invalid states:
  - FWCore/PythonParameterSet/BuildFile.xml:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/makeprocess_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/makepset_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/BuildFile.xml:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/python/test/testTask_cff.py:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/ufip.txt:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/fip.txt:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/processbuilder_t.cppunit.cpp:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/readpsetsfrom_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f

cmsbuild commented:
A new Pull Request was created by  @davidlange6 (David Lange) for master.

It involves the following packages:

FWCore/Framework
FWCore/Integration
FWCore/ParameterSetReader
FWCore/PyDevParameterSet
FWCore/PythonFramework
FWCore/PythonParameterSet
Fireworks/FWInterface


@cmsbuild, @smuzaffar, @Dr15Jones, @alja can you please review it and eventually sign? Thanks.
@makortel, @alja, @wddgit, @Martin-Grunewald this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

davidlange6 commented:
please test




On 6 Apr 2019, at 20:31, cmsbuild <notifications@github.com<mailto:notifications@github.com>> wrote:


+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26377/9122

  *   This PR adds an extra 56KB to repository

  *   Found files with invalid states:

     *   FWCore/PythonParameterSet/BuildFile.xml:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>
     *   FWCore/PythonParameterSet/test/makeprocess_t.cppunit.cc<http://makeprocess_t.cppunit.cc>:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>
     *   FWCore/PythonParameterSet/test/makepset_t.cppunit.cc<http://makepset_t.cppunit.cc>:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>
     *   FWCore/PythonParameterSet/test/BuildFile.xml:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>
     *   FWCore/PythonParameterSet/python/test/testTask_cff.py:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>
     *   FWCore/PythonParameterSet/test/ufip.txt:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>
     *   FWCore/PythonParameterSet/test/fip.txt:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>
     *   FWCore/PythonParameterSet/test/processbuilder_t.cppunit.cpp:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>
     *   FWCore/PythonParameterSet/test/readpsetsfrom_t.cppunit.cc<http://readpsetsfrom_t.cppunit.cc>:
        *   Deleted: cc6db46<https://github.com/cms-sw/cmssw/commit/cc6db462bf443e24b681eaf69f1ee13b60d4b506>
        *   Added: 9e22e8d<https://github.com/cms-sw/cmssw/commit/9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/26377#issuecomment-480526870>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEzyw01G6RSguRkBtKPzRc0lHcS775Qfks5veOfNgaJpZM4cgT0w>.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: fef2af57fb76b3172b271954cc572db162bad238
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26377/34037/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26377/34037/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3140495
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3140297
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

Dr15Jones commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented on FWCore/PyDevParameterSet/BuildFile.xml:
```diff
@@ -2,7 +2,7 @@
 <use   name="FWCore/Utilities"/>
 <use   name="py2-pybind11"/>
-<use   name="python"/>
+<use   name="python3"/>
```
ok, so whatever uses PyDevParameterSet from now onwards is effectively bound to python3

fabiocos commented on FWCore/PyDevParameterSet/interface/PyBind11ProcessDesc.h:
```diff
@@ -13,6 +13,8 @@ namespace edm {
   class ProcessDesc;
 }
 
+namespace cmspython3 {
```
this namespace is practically creating a parallel world for the python3 version of the framework interpreter. I assume this is for the transition period, asymptotically this will converge back to edm, am I correct?

fabiocos commented:
@davidlange6 @Dr15Jones I see that the edmConfigHash and edmParameterSetDump utilities are gone. Is this desired? As far as I can see they could be reimplemented by moving boost_python to pybind11. Am I missing something?

fabiocos commented:
I have done a small test in https://github.com/fabiocos/cmssw/pull/new/fc-prova and it seems to work

Dr15Jones commented on FWCore/PyDevParameterSet/src/PyBind11ProcessDesc.cc:
```diff
@@ -9,108 +9,110 @@
+    edm::python3::initializePyBind11Module();
+    prepareToRead();
+    {
+#if PY_MAJOR_VERSION >= 3
```
Is this needed anymore since this package is now Python3 only?

Dr15Jones commented:
In my hasty review I missed the fact that two helper executables were removed. edmConfigHash may be used by workflow management.

davidlange6 commented:
I just missed the bin area (is a strange place for it perhaps) will restore it

David


On 7 Apr 2019, at 15:52, Chris Jones <notifications@github.com<mailto:notifications@github.com>> wrote:


In my hasty review I missed the fact that two helper executables were removed. edmConfigHash may be used by workflow management.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/26377#issuecomment-480592367>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEzyw7s-IX8_PXMEIg0kddWTgiZwVNItks5veff1gaJpZM4cgT0w>.

davidlange6 committed:
restore lost execs

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-26377/9128

- This PR adds an extra 60KB to repository

- Found files with invalid states:
  - FWCore/PythonParameterSet/BuildFile.xml:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/makeprocess_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/makepset_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/BuildFile.xml:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/python/test/testTask_cff.py:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/ufip.txt:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/fip.txt:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/processbuilder_t.cppunit.cpp:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f
  - FWCore/PythonParameterSet/test/readpsetsfrom_t.cppunit.cc:
    - Deleted: cc6db462bf443e24b681eaf69f1ee13b60d4b506
    - Added: 9e22e8d5821b54b9cbd1b2b5e5516c6d80c7fa6f

cmsbuild commented:
Pull request #26377 was updated. @cmsbuild, @smuzaffar, @Dr15Jones, @alja can you please check and sign again.

fabiocos commented on FWCore/ParameterSetReader/bin/edmConfigHash.cpp:
```diff
@@ -29,7 +29,7 @@ int main(int argc, char **argv) try {
   }
 
-  std::shared_ptr<edm::ParameterSet> parameterSet = edm::boost_python::readConfig(config);
+  std::shared_ptr<edm::ParameterSet> parameterSet = edm::readConfig(config);
```
ok, pybind11 is already inside ParameterSetReader, so no need to call it explicitly...

fabiocos commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

davidlange6 commented:
yes, that is why I moved these two tools to FWCore/ParameterSetReader. They should be derived from whatever the default system CMSSW uses for its parameter set interface.

> On Apr 7, 2019, at 6:51 PM, Fabio Cossutti <notifications@github.com> wrote:
> 
> @fabiocos commented on this pull request.
> 
> In FWCore/ParameterSetReader/bin/edmConfigHash.cpp:
> 
> > @@ -29,7 +29,7 @@ int main(int argc, char **argv) try {
>      config = std::string(argv[1]);
>    }
>  
> -  std::shared_ptr<edm::ParameterSet> parameterSet = edm::boost_python::readConfig(config);
> +  std::shared_ptr<edm::ParameterSet> parameterSet = edm::readConfig(config);
> 
> ok, pybind11 is already inside ParameterSetReader, so no need to call it explicitly...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmsbuild commented:
+1
Tested at: 8971438d99f76598c61a23e1f3d1ee0bf95ec419
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26377/34040/summary.html

cmsbuild commented:
Comparison job queued.

fabiocos commented:
@Dr15Jones the latest addition reintroduce the two utilities, the rest is unchanged. Do you confirm the signature?

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26377/34040/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 32
 - DQMHistoTests: Total histograms compared: 3140495
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3140297
 - DQMHistoTests: Total skipped: 197
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 31 files compared)
 - Checked 133 log files, 14 edm output root files, 32 DQM output files

Dr15Jones commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

fabiocos commented:
+1
